config/config.go

```diff
@@ -148,7 +148,7 @@ type Config struct {
 	Analytics AnalyticsConfig `yaml:"analytics,omitempty" json:"analytics,omitempty" mapstructure:"analytics,omitempty"`
 	// If EnvConfigs is specified, APIKeyHeader, APIKeyClaim, JWTProviderKey in AuthConfig will be ineffectual.
 	Auth       AuthConfig `yaml:"auth,omitempty" json:"auth,omitempty" mapstructure:"auth,omitempty"`
-	EnvConfigs EnvConfigs `yaml:"env_configs,omitempty" json:"env_configs,omitempty" mapstructure:"env_configs,omitempty"`
+	Environments EnvironmentConfigs `yaml:"environments,omitempty" json:"environments,omitempty" mapstructure:"environments,omitempty"`
```

Comment:
The term "Environment" is so overloaded at this point - and this isn't even related to the existing one. Can we come up with something that adds clarity? APIGroup? ProxySet? OperationVersions? Some combination of those words, or...?

Reply:
`ApigeeEnvironments`? That's what this represents after all. Are the other references to `Environment` not also an Apigee Environment?

Reply:
But this isn't, really, is it? It's more like a Environment Revision or Version. I'm just trying to make this make sense for users and this doesn't have any reference to the Apigee Environment we're operating in (ie. the runtime we're making requests to). So, we have a `Config.Global.Environment` and accept `apigee_environment` from Envoy as metadata to override that. Now we're adding another Environment to the config... it's confusing.

Reply:
I think this comes down to domain entity mapping. The domain is obviously Apigee, thus I am very reluctant to add a brand new entity (e.g. "API Group"). 

FWIW I could not find any references to an Apigee Environment in config.go, the closest is `TenantConfig.EnvName`, is that what you mean?

Reply:
Ah I see at least part of the confusion.... this is probably N configs for the _same_ Environment, thus the "revisions" concept. So what about just `EnvironmentConfigs`?

Reply:
Oops... sorry, yes. I meant `TenantConfig.EnvName`.

Reply:
Changed to `EnvironmentConfigs` and updating comments to explain the semantics.